contracts/extensions/VotingReputation.sol

```diff
@@ -0,0 +1,893 @@
+    tokenLocking.transfer(token, voterReward, msg.sender, true);
+
+    emit PollVoteRevealed(_pollId, msg.sender, _vote);
+
+    if (poll.repRevealed == poll.repSubmitted) {
+      poll.events[REVEAL_END] = uint64(now);
+
+      emit PollEventSet(_pollId, REVEAL_END);
+    }
+  }
+
+  /// @notice Escalate a poll to a higher domain
+  /// @param _pollId The id of the poll
+  /// @param _newDomainId The desired domain of escalation
+  /// @param _childSkillIndex For the current domain, relative to the escalated domain
+  /// @param _key Reputation tree key for the new domain
+  /// @param _value Reputation tree value for the new domain
+  /// @param _branchMask The branchmask of the proof
+  /// @param _siblings The siblings of the proof
+  function escalatePoll(
+    uint256 _pollId,
+    uint256 _newDomainId,
+    uint256 _childSkillIndex,
+    bytes memory _key,
+    bytes memory _value,
+    uint256 _branchMask,
+    bytes32[] memory _siblings
+  )
+    public
+  {
+    Poll storage poll = polls[_pollId];
+    require(getPollState(_pollId) == PollState.Closed, "voting-rep-poll-not-closed");
+
+    uint256 newDomainSkillId = colony.getDomain(_newDomainId).skillId;
+    uint256 childSkillId = colonyNetwork.getChildSkillId(newDomainSkillId, _childSkillIndex);
+    require(childSkillId == poll.skillId, "voting-rep-invalid-domain-proof");
+
+    delete poll.events;
+
+    poll.events[STAKE_END] = uint64(now + stakePeriod);
+    poll.events[SUBMIT_END] = uint64(now + stakePeriod + submitPeriod);
+    poll.events[REVEAL_END] = uint64(now + stakePeriod + submitPeriod + revealPeriod);
+
+    poll.domainId = _newDomainId;
+    poll.skillId = newDomainSkillId;
+    poll.skillRep = getReputationFromProof(_pollId, address(0x0), _key, _value, _branchMask, _siblings);
+
+    emit PollEscalated(_pollId, msg.sender, _newDomainId);
+  }
+
+  function finalizePoll(uint256 _pollId) public {
+    Poll storage poll = polls[_pollId];
+    require(getPollState(_pollId) == PollState.Finalizable, "voting-rep-poll-not-executable");
+
+    poll.finalized = true;
+
+    bool canExecute = (
+      poll.stakes[NAY] <= poll.stakes[YAY] &&
+      poll.votes[NAY] <= poll.votes[YAY]
+    );
+
+    if (getSig(poll.action) == CHANGE_FUNCTION) {
+      bytes32 structHash = hashExpenditureStruct(poll.action);
+      expenditurePollCounts[structHash] = sub(expenditurePollCounts[structHash], 1);
+
+      // Release the claimDelay if this is the last active poll
+      if (expenditurePollCounts[structHash] == 0) {
+        bytes memory claimDelayAction = createClaimDelayAction(poll.action, 0);
+        require(executeCall(_pollId, claimDelayAction), "voting-rep-expenditure-unlock-failed");
+      }
+
+      uint256 requiredStake = getRequiredStake(_pollId);
+      uint256 votePower = (poll.stakes[NAY] < requiredStake) ? poll.stakes[YAY] : poll.votes[YAY];
+      bytes32 slotHash = hashExpenditureSlot(poll.action);
+
+      if (expenditurePastPolls[slotHash] < votePower) {
```

Comment:
The physicist in me is one again raising eyebrows. `votePower` can be either be an amount of tokens (if `NAY` did not stake enough tokens) or an amount of reputation that voted (if `NAY` did stake enough tokens), and doing a direct comparison between things with different units surely leads to weird behaviour?

 If `YAY` passed a vote, it was escalated, and then no-one staked further, `poll.stakes[YAY]` could easily be less than the reputation that voted in the lower domain, which would cause this to not able to be executed... is that desirable?

Reply:
It's a fair point, but I think we're ok inasmuch as the token-denominated stake is ultimately a function of reputation -- if someone staked 100 tokens, then they have 100 reputation, and they would have ostensibly voted with that reputation.

I think it's reasonable for a no-contest motion to not have sufficient power to override a decision in a lower domain, if that's how the reputation works. The motion-maker should have chosen a higher domain, such that the stake would have been sufficient to override the previous vote.

Reply:
> I think it's reasonable for a no-contest motion to not have sufficient power to override a decision in a lower domain, if that's how the reputation works.

Right, and I'm saying that's not what is happening here.

YAY won the vote in the lower domain, and everyone with reputation voted. We escalate, and no-one further stakes on either side, and this function is called. `votePower` becomes `poll.stakes[YAY]`, which is almost assuredly less than the reputation that voted in the lower domain. `canExecute` becomes false in that scenario, does it not, despite YAY winning in the lower domain and no vote occurring in the higher domain?

> If YAY passed a vote, it was escalated, and then no-one staked further, poll.stakes[YAY] could easily be less than the reputation that voted in the lower domain, which would cause this to not able to be executed.

Reply:
Ah I see your point. In my mind we also checked for prior votes to determine whether to use votes or stakes.

Let's do it like:

```
uint256 votePower = (poll.votes[YAY] < 0) ? poll.votes[YAY] : poll.stakes[YAY];
```

That way the condition implicitly includes whether or not the staking has occurred.

Reply:
We'll need to check `poll.votes[NAY]` as well to avoid going wrong in the case where noone voted for `YAY` (in which case, the weight should still be `poll.votes[YAY]` and not `poll.stakes[YAY]`) but otherwise I think this works.

Reply:
Fair enough